src/transport/SessionManager.h

```diff
@@ -174,8 +174,155 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate, public FabricTabl
     Optional<SessionHandle> AllocateSession(Transport::SecureSession::Type secureSessionType,
                                             const ScopedNodeId & sessionEvictionHint);
 
+    /**
+     * @brief
+     *  A set of templated helper function that call a provided lambda
+     *  on all sessions in the underlying session table that match the provided
+     *  query criteria.
+     *
+     */
+
+    /**
+     * @brief
+     *
+     * Call the provided lambda on sessions that match the provided ScopedNodeId.
+     *
+     */
+    template <typename Function>
+    void ForEachMatchingSession(const ScopedNodeId & node, Function && function)
+    {
+        mSecureSessions.ForEachSession([&](auto * session) {
+            if (session->GetPeer() == node)
+            {
+                function(session);
+            }
+
+            return Loop::Continue;
+        });
+    }
+
+    /**
+     * @brief
+     *
+     * Call the provided lambda on sessions that match the provided fabric index.
+     *
+     */
+    template <typename Function>
+    void ForEachMatchingSession(FabricIndex fabricIndex, Function && function)
+    {
+        mSecureSessions.ForEachSession([&](auto * session) {
+            if (session->GetFabricIndex() == fabricIndex)
+            {
+                function(session);
+            }
+
+            return Loop::Continue;
+        });
+    }
+
+    /**
+     * @brief
+     *
+     * Call the provided lambda on all sessions that match the logical fabric
+     * associated with the provided ScopedNodeId and target the same logical node.
+     *
+     */
+    template <typename Function>
+    CHIP_ERROR ForEachCollidingSession(const ScopedNodeId & node, Function && function)
```

Comment:
Why "colliding". Colliding on what ? WHy not just `ForEachSessionMatchingScopedNode` ? "Colliding" is pejorative and seems to indicate abnormal condition in the name, versus just finding matches

Reply:
Sure. But the suggested name is not much clearer in conveying a match on the _underlying_ logical fabric. 

I can rename it to `ForEachSessionMatchingLogicalFabric`.

Reply:
See [this](https://github.com/project-chip/connectedhomeip/pull/21420#discussion_r933396241) comment from @msandstedt for context as well.

Reply:
I've re-named these to + added comments:

```
void ForEachMatchingSession(const ScopedNodeId & node, Function && function)
void ForEachMatchingSession(FabricIndex fabricIndex, Function && function)
CHIP_ERROR ForEachSessionOnLogicalFabric(const ScopedNodeId & node, Function && function)
CHIP_ERROR ForEachSessionOnLogicalFabric(FabricIndex fabricIndex, Function && function)

void ExpireAllSessions(const ScopedNodeId & node);
void ExpireAllSessionsForFabric(FabricIndex fabricIndex);
CHIP_ERROR ExpireAllSessionsOnLogicalFabric(const ScopedNodeId & node);
CHIP_ERROR ExpireAllSessionsOnLogicalFabric(FabricIndex fabricIndex);
```